### PR TITLE
Extended just dance dataset + small fixes

### PR DESCRIPTION
**Pull request by Nadav-Barak**

### CODE REVIEW

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py [hunk 1]
```diff
@@ -28,26 +28,132 @@
...
+    use_full_size : bool, default: False
+        If True, the returned data will be the full dataset, otherwise returns a subset of the data.
+    Returns
+    -------
+    predictions : np.ndarray
+        The prediction of the data elements in the dataset.
+
+    """
+    if use_full_size:
+        raise NotImplementedError('Predictions for the full dataset are not yet available.')
```
Comment (yromanyshyn) [inline deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:78]: why is this parameter then even needed? maybe drop it?

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py [hunk 2]
```diff
@@ -63,6 +62,10 @@ def run_logic(self, context: Context, dataset_kind) -> CheckResult:
         # Decide which scorer and score_per_sample to use in the algorithm run
         is_multilabel = text_data.is_multi_label_classification()
         if is_multilabel:
+            if self.alternative_scorer is None:
+                self.alternative_scorer = {'Precision Macro': 'precision_macro'}
```
Comment (noamzbr) [inline deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:66]: why not f1? precision optimizes for lowering FP while our loss is more "balanced" between FP and FN, which may create a situation in which a weak segment is detected that has a higher / equivalent precision compared to baseline, while it's loss is higher.

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py (hunk 1 above)
Comment (noamzbr) [reply deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:78]: For compatibility with the other loading functions in this file I would imagine. 
@Nadav-Barak I would think though that the reasonable default for use_full_data is true.

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py [hunk 3]
```diff
@@ -28,26 +28,132 @@
...
+        If True, the returned data is split into train and test exactly like the toy model
+        was trained. The first return value is the train data and the second is the test data.
+        Otherwise, returns a single object.
+    use_full_size : bool, default: False
+        If True, the returned data will be the full dataset, otherwise returns a subset of the data.
+
+    Returns
+    -------
+    embeddings : np.ndarray
+        Embeddings for the tweet_emotion dataset.
```
Comment (noamzbr) [inline deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:111]: Not a tweet. Do ctrl+f for "tweet" in this file there are a couple of docstrings that need updating.

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py [hunk 4]
```diff
@@ -75,27 +185,28 @@ def load_data(data_format: str = 'TextData', as_train_test: bool = True, use_ful
...
         return dataset
 
     else:
-        if use_full_size:
-            train = data[data['dateComment'] < '2015-01-01']
-            test = data[data['dateComment'] >= '2015-01-01']
-        else:
-            train = data[data['dateComment'] < '2013-01-01']
-            test = data[data['dateComment'] >= '2021-01-01']
+        train = data[data[_TIME_COL] < _DATE_TO_SPLIT_BY]
```
Comment (noamzbr) [inline deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:208]: can't we use _get_train_test_indexes here as well?

File: deepchecks/nlp/utils/text_properties.py [hunk 5]
```diff
@@ -282,7 +282,7 @@ def _predict(text, classifier, kind):
             return v['score']
         elif kind == 'fluency':
             label_value = 'LABEL_1'
-        elif kind == 'fluency':
+        elif kind == 'formality':
```
Comment (noamzbr) [inline deepchecks/nlp/utils/text_properties.py:285]: I know this is not related to this PR but I regret this code and think the treatment of prediction for each property should be within each property, as in this implementation the logic is split between two functions and it's confusing...

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py (hunk 1 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:78]: The partial size is good enough, 15K samples

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py (hunk 4 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:208]: we can, will update

File: deepchecks/nlp/utils/text_properties.py (hunk 5 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/utils/text_properties.py:285]: i can change it none the less, any objections @yromanyshyn ?

File: deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py (hunk 2 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/checks/model_evaluation/weak_segments_performance.py:66]: Not a strong preference, changed

File: deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py (hunk 3 above)
Comment (Nadav-Barak) [reply deepchecks/nlp/datasets/classification/just_dance_comment_analysis.py:111]: you right, done

**Review by noamzbr — APPROVED** [review]

**Pull request merged**